llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -244,7 +244,8 @@ class DependencyGraph {
         Graph[UFunc].insert(V);
       } else if (isa<const Constant>(U)) {
         if (const auto *GV = dyn_cast<const GlobalVariable>(U))
-          Graph[GV].insert(V);
+          if (Graph[GV].insert(V).second == false)
```

Comment:
can you explain this change? it's unclear why its needed from the title and description. thanks

Reply:
You are right, the compiler invocation that needs this change is not tested by this PR, so I'll revert this change for now. I'll open a subsequent separate PR for this. Thank you.

Reply:
This is now reverted